node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/NodePrioritizer.java

```diff
@@ -71,9 +71,14 @@
                 .filter(clusterSpec.group()::equals)
                 .count();
 
-        this.allocateForReplacement = isReplacement(nodesInCluster.size(),
-                                                    nodesInCluster.state(Node.State.failed).size());
-        this.canAllocateNew = requestedNodes instanceof NodeSpec.CountNodeSpec;
+        // In dynamically provisioned zones, we can always take spare hosts since we can provision new on-demand,
+        // NodeCandidate::compareTo will ensure that they will not be used until there is no room elsewhere.
+        // In non-dynamically provisioned zones, we only allow allocating to spare hosts to replace failed nodes.
+        this.canAllocateToSpaceHosts = isReplacement(
+                nodesInCluster.size(), nodesInCluster.state(Node.State.failed).size()) || dynamicProvisioning;
```

Comment:
Nit:  put dynamicProvisioning as the first condition of the `||` to short-circuit isReplacement expression if true